### PR TITLE
[Refactoring] Fix crasher in switch with non-nominal subject type

### PR DESCRIPTION
**Pull request by rintaro**
Added nullptr check.
rdar://problem/42098130

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/Refactoring.cpp [hunk 1]
```diff
@@ -2393,7 +2393,10 @@ collectAvailableRefactoringsAtCursor(SourceFile *SF, unsigned Line,
 
 static EnumDecl* getEnumDeclFromSwitchStmt(SwitchStmt *SwitchS) {
   if (auto SubjectTy = SwitchS->getSubjectExpr()->getType()) {
-    return SubjectTy->getAnyNominal()->getAsEnumOrEnumExtensionContext();
```
Comment (slavapestov) [inline lib/IDE/Refactoring.cpp]: Simpler: `dyn_cast<EnumDecl>(SubjectTy->getAnyNominal())`

**Review by nkcsgexi — APPROVED** [review]
Lgtm!

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**